qemu/tests/numa_prealloc_threads.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+from virttest import env_process
+from virttest import error_context
+from virttest.qemu_monitor import QMPCmdError
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    numa_prealloc_threads test
+    1) Boot a guest with thread-context and cpu-affinity option
+    2) Obtain the thread-id
+    3) Check the affinity obtained from QEMU is correct
+    4) With sandbox enabled, try to change the cpu-affinity
+       and handle the error
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    test.log.info("Check host CPUs number")
+    host_cpus = int(process.system_output("nproc"))
+    smp_fixed = params.get_numeric("smp_fixed")
+    if host_cpus < smp_fixed:
+        test.cancel("The host only has %d CPUs, it needs at least %d!" % (host_cpus, smp_fixed))
+
+    params['not_preprocess'] = "no"
+    env_process.preprocess(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login()
+
+    device = "thread_context-" + params["vm_thread_contexts"]
+    error_msg = params.get("sandbox_error_message", "")
+
+    thread_id = vm.monitor.qom_get(device, "thread-id")
+    if not thread_id:
+        test.fail("No thread-id setted.")
+
+    test_affinity = params.get("vm_thread_context_cpu-affinity")
+    cpu_affinity = vm.monitor.qom_get(device, "cpu-affinity")
+    affinity = str(cpu_affinity[0]) + "-" + str(cpu_affinity[-1])
+    test.log.debug("The affinity: %s and the test_affinity: %s"
+                   % (affinity, test_affinity))
+    if test_affinity != affinity:
+        test.fail("Test and QEMU cpu-affinity does not match!")
+
+    cmd_taskset = "taskset -c -p " + str(thread_id)
+    output = process.system_output(cmd_taskset)
```

Comment:
`process.getoutput` will return string instead of `bytes`

Reply:
Updated now, thanks!